app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -86,6 +86,13 @@ class KiwixMainActivity : CoreMainActivity() {
     }
   }
 
+  override fun onPostCreate(savedInstanceState: Bundle?) {
```

Comment:
similarly here, avoid onPostCreate

Reply:
and I think this entire handling may be wrong, why isn't the nav library helping you deeplink? https://developer.android.com/guide/navigation/navigation-deep-link

Reply:
Can't get the deep link to navigate to any fragment for some reason, will try some more later..